projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy_optimise.php

```diff
@@ -258,149 +269,167 @@ static function shorthand($value) {
 	 */
-	function compress_numbers($subvalue) {
-		$unit_values = & $GLOBALS['csstidy']['unit_values'];
+	public function compress_numbers( $subvalue ) {
+		$unit_values  = & $GLOBALS['csstidy']['unit_values'];
 		$color_values = & $GLOBALS['csstidy']['color_values'];
 
-		// for font:1em/1em sans-serif...;
-		if ($this->property === 'font') {
-			$temp = explode('/', $subvalue);
+		// for font:1em/1em sans-serif...;.
+		if ( $this->property === 'font' ) {
+			$temp = explode( '/', $subvalue );
 		} else {
-			$temp = array($subvalue);
+			$temp = array( $subvalue );
 		}
-		for ($l = 0; $l < count($temp); $l++) {
-			// if we are not dealing with a number at this point, do not optimise anything
-			$number = $this->AnalyseCssNumber($temp[$l]);
-			if ($number === false) {
+		for ( $l = 0; $l < count( $temp ); $l++ ) { // phpcs:ignore Generic.CodeAnalysis.ForLoopWithTestFunctionCall.NotAllowed, Squiz.PHP.DisallowSizeFunctionsInLoops.Found
+			// if we are not dealing with a number at this point, do not optimise anything.
+			$number = $this->analyse_css_number( $temp[ $l ] );
+			if ( $number === false ) {
 				return $subvalue;
 			}
 
-			// Fix bad colors
-			if (in_array($this->property, $color_values)) {
-				if (strlen($temp[$l]) == 3 || strlen($temp[$l]) == 6) {
-					$temp[$l] = '#' . $temp[$l];
-				}
-				else {
-					$temp[$l] = "0";
+			// Fix bad colors.
+			if ( in_array( $this->property, $color_values ) ) { // phpcs:ignore WordPress.PHP.StrictInArray.MissingTrueStrict
+				if ( strlen( $temp[ $l ] ) == 3 || strlen( $temp[ $l ] ) == 6 ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
+					$temp[ $l ] = '#' . $temp[ $l ];
+				} else {
+					$temp[ $l ] = '0';
 				}
 				continue;
 			}
 
-			if (abs($number[0]) > 0) {
-				if ($number[1] == '' && in_array($this->property, $unit_values, true)) {
+			if ( abs( $number[0] ) > 0 ) {
+				if ( $number[1] == '' && in_array( $this->property, $unit_values, true ) ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
 					$number[1] = 'px';
 				}
 			} else {
 				$number[1] = '';
 			}
 
-			$temp[$l] = $number[0] . $number[1];
+			$temp[ $l ] = $number[0] . $number[1];
 		}
 
-		return ((count($temp) > 1) ? $temp[0] . '/' . $temp[1] : $temp[0]);
+		return ( ( count( $temp ) > 1 ) ? $temp[0] . '/' . $temp[1] : $temp[0] );
 	}
 
 	/**
 	 * Checks if a given string is a CSS valid number. If it is,
 	 * an array containing the value and unit is returned
-	 * @param string $string
+	 *
+	 * @param string $string - the string we're checking.
 	 * @return array ('unit' if unit is found or '' if no unit exists, number value) or false if no number
 	 */
-	function AnalyseCssNumber($string) {
+	public function analyse_css_number( $string ) {
 		// most simple checks first
-		if (strlen($string) == 0 || ctype_alpha($string[0])) {
+		if ( strlen( $string ) == 0 || ctype_alpha( $string[0] ) ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
```

Comment:
If `$string` is anything that would make `strlen()` return null in PHP 5.6 or 7, it would throw a fatal error in PHP 8.